### PR TITLE
Bash completion

### PR DESCRIPTION
**Pull request by vanadinit**

### CODE REVIEW

File: automatix/bash_completion.py [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+
+class ScriptFileCompleter(object):
+    """
+    Scriptfile completer
+    """
+    def __init__(self, script_path: str):
+        self.script_path = script_path
+
+    def __call__(self, prefix, **kwargs):
+        completion = []
```
Comment (j-p-a-u-l) [inline automatix/bash_completion.py:20]:
```suggestion
        completion = []
        pre_len = len(self.script_path)+1
```

File: automatix/bash_completion.py [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+        directories = _call(["bash", "-c", f"compgen -A directory -- '{self.script_path}/{prefix}'"])
+        completion += [f'{d}/' for d in directories]
```
Comment (j-p-a-u-l) [inline automatix/bash_completion.py:22]:
```suggestion
        completion += [f'{d[pre_len:]}/' for d in directories]
```

File: automatix/bash_completion.py [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+        for ext in ['yaml', 'yml']:
+            files = _call(["bash", "-c", f"compgen -A file -X '!*.{ext}' -- '{self.script_path}/{prefix}'"])
+            completion += [f[len(self.script_path)+1:] for f in files]
```
Comment (j-p-a-u-l) [inline automatix/bash_completion.py:25]:
```suggestion
            completion += [f[pre_len:] for f in files]
```

**Review by j-p-a-u-l — COMMENTED** [review]
Takes long time on my setup ...
Don't know why yet ...

**Review by nkoch-ops — APPROVED** [review]
Nice improvement. Works well for me, even on fish.

**Comment by j-p-a-u-l** [comment]
This will be included in PR #38

**Pull request closed without merging**